packages/react-instantsearch-core/src/connectors/__tests__/connectQueryRules.ts

```diff
@@ -67,6 +76,354 @@ describe('connectQueryRules', () => {
         ]);
       });
     });
+
+    describe('trackedFilters', () => {
+      it('does not set ruleContexts without search state and trackedFilters', () => {
+        const props: QueryRulesProps = {
+          ...defaultProps,
+        };
+        const searchState = {};
+        const searchParameters = getSearchParameters(
+          new SearchParameters(),
+          props,
+          searchState
+        );
+
+        expect(searchParameters.ruleContexts).toEqual(undefined);
+      });
+
+      it('does not set ruleContexts with search state but without tracked filters', () => {
+        const props: QueryRulesProps = {
+          ...defaultProps,
+        };
+        const searchState = {
+          range: {
+            price: {
+              min: 20,
+              max: 3000,
+            },
+          },
+        };
+        const searchParameters = getSearchParameters(
+          new SearchParameters(),
+          props,
+          searchState
+        );
+
+        expect(searchParameters.ruleContexts).toEqual(undefined);
+      });
+
+      it('does not reset initial ruleContexts with trackedFilters', () => {
+        const props: QueryRulesProps = {
+          ...defaultProps,
+          trackedFilters: {
+            price: values => values,
+          },
+        };
+        const searchState = {};
+        const searchParameters = getSearchParameters(
+          SearchParameters.make({
+            ruleContexts: ['initial-rule'],
+          }),
+          props,
+          searchState
+        );
+
+        expect(searchParameters.ruleContexts).toEqual(['initial-rule']);
+      });
+
+      it('sets ruleContexts based on range', () => {
+        const priceSpy = jest.fn(values => values);
+        const props: QueryRulesProps = {
+          ...defaultProps,
+          trackedFilters: {
+            price: priceSpy,
+          },
+        };
+        const searchState = {
+          range: {
+            price: {
+              min: 20,
+              max: 3000,
+            },
+          },
+        };
+        const searchParameters = getSearchParameters(
+          new SearchParameters(),
+          props,
+          searchState
+        );
+
+        expect(priceSpy).toHaveBeenCalledTimes(1);
+        expect(priceSpy).toHaveBeenCalledWith([20, 3000]);
+        expect(searchParameters.ruleContexts).toEqual([
+          'ais-price-20',
```

Comment:
This format is expected. The one for `hierarchicalMenu` is different in InstantSearch.js. This is the best we can do in RIS because there's no way to get the separator value. SEs can still transform the rule with `transformRuleContexts` if needed.